src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionListProvider.cs

```diff
@@ -112,6 +118,24 @@ internal class DelegatedCompletionListProvider
         return rewrittenResponse;
     }
 
+    private async Task<bool> ShouldIncludeSnippetsAsync(VersionedDocumentContext documentContext, int absoluteIndex, CancellationToken cancellationToken)
+    {
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
❓ is this something we should be doing on the client side instead? It's client specific logic, but idk the best way to go from `TextDocumentIdentifier` to `RazorCodeDocument` without causing some reparse or worse.